src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1782,30 +1782,32 @@ class FunctionBinaryArithmeticWithConstants : public FunctionBinaryArithmetic<Op
                 {
                     ColumnsWithTypeAndName columns_with_constant
                         = {{left.column->cloneResized(1), left.type, left.name},
-                           {right.type->createColumnConst(1, point), right.type, right.name}};
+                           {removeNullable(right.type)->createColumnConst(1, point), removeNullable(right.type), right.name}};
```

Comment:
Maybe do `removeNullable(recursiveRemoveLowCardinality(right.type))`?

Reply:
Yes, totally. Fuzzer found it as well :)